File: E-Commerce/src/app/shared/models/product.ts
import { User } from './user';

export interface Product {
  id: number;
  title: string;
  description: string;
  productCategory: Category;
  offer: Offer;
  price: number;
  quantity: number;
  imageName: string;
}

export interface Offer {
  id: number;
  title: string;
  discount: number;
}

export interface Category {
  id: number;
  category: string;
  subCategory: string;
}

export interface Review {
  id: number;
  user: User;
  product: Product;
  value: string;
  createdAt: string;
}

File: E-CommerceAPI/E-CommerceAPI/Models/Product.cs
using System;
using System.Collections.Generic;

namespace E_CommerceAPI.Models;

public partial class Product
{
    public int ProductId { get; set; }

    public string Title { get; set; } = null!;

    public string Description { get; set; } = null!;

    public int CategoryId { get; set; }

    public int OfferId { get; set; }

    public double Price { get; set; }

    public int Quantity { get; set; }

    public string ImageName { get; set; } = null!;

    public virtual ProductCategory Category { get; set; } = null!;

    public virtual Offer Offer { get; set; } = null!;
}
